BraveWallet/WalletProviderPermissionRequestsManager.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveCore
+import Data
+
+/// A permission request for a specific dapp
+public struct WebpagePermissionRequest: Equatable {
+  /// A users response type
+  public enum Response {
+    /// The user rejected the prompt by dismissing the screen
+    case rejected
+    /// The user granted access to a set of accounts
+    case granted(accounts: [String])
+  }
+  /// The origin that requested this permission
+  let requestingOrigin: URL
+  /// The type of request
+  let coinType: BraveWallet.CoinType
```

Comment:
This is only for connection requests, since we can't use the Chromium permissions system. Coin type is added here since we will eventually add Solana and others

Reply:
Ok, it doesn't seem to present any security concerns for me at this point so no further action would be needed on this one.